library/alloc/src/alloc.rs

```diff
@@ -30,8 +30,10 @@ extern "Rust" {
     #[rustc_allocator]
     #[rustc_allocator_nounwind]
     fn __rust_alloc(size: usize, align: usize) -> *mut u8;
```

Comment:
We should also annotate the alloc_zeroed below (with zeroed allockind).

Reply:
Done, but I'm uneasy per a comment in a bug about "weird UB-like behaviour with bitvec iterators in rustc_data_structures" last time we did this. It's a separate commit so it'll be easy to back it out if we see the problem, but maybe we should even do it as a separate PR to isolate possible blast radius?

Reply:
I'm not particularly concerned about this, given how much time has passed. I have been testing rust with `__rust_alloc_zeroed` registered as a TLI allocator (before your patches...) and didn't encounter any issues.